src/main/java/com/imaee/propinq/buildings/mappers/BuildingMapper.java

```diff
@@ -0,0 +1,52 @@
+package com.imaee.propinq.buildings.mappers;
+
+import com.imaee.propinq.buildings.controllers.responses.BuildingResponse;
+import com.imaee.propinq.buildings.data.models.Building;
+import com.imaee.propinq.buildings.data.models.BuildingType;
+import com.imaee.propinq.shared.data.models.Review;
+import com.imaee.propinq.shared.data.models.Image;
+import com.imaee.propinq.properties.data.models.Property;
+import com.imaee.propinq.users.data.models.User;
+
+import java.util.List;
+import java.util.UUID;
+
+public class BuildingMapper {
+
+    private BuildingMapper() {}
+
+    public static BuildingResponse toBuildingResponse(Building building) {
+        List<String> imagesURLS = getImagesURLs(building);
+        UUID userId = building.getUser().getUserId();
+        UUID buildingTypeId = building.getBuildingType().getBuildingTypeId();
+        return new BuildingResponse(
+                building.getBuildingId(),
+                building.getName(),
+                building.getDescription(),
+                building.getAddress(),
+                imagesURLS,
+                userId,
+                buildingTypeId
+        );
+    }
```

Comment:
La responsabilidad de los _mappers_ es únicamente construir objetos. No deben actuar como recuperadores de información: para eso están los **services**. En definitiva: los métodos que estén declarados en cualquier mapper deben tener TODA la información necesaria para construir los objetos

```suggestion
    public static BuildingResponse toBuildingResponse(Building building, List<String> 
            imagesURLS, UUID userId, UUID buildingTypeId) {
        return new BuildingResponse(
                building.getBuildingId(),
                building.getName(),
                building.getDescription(),
                building.getAddress(),
                imagesURLS,
                userId,
                buildingTypeId
        );
    }
```